hazelcast/src/main/java/com/hazelcast/client/properties/ClientProperty.java

```diff
@@ -118,7 +118,7 @@
      * <p>Discovery SPI is <b>disabled</b> by default</p>
      */
     public static final HazelcastProperty DISCOVERY_SPI_PUBLIC_IP_ENABLED
```

Comment:
Shouldn't we update the javadoc for this property? The current one duplicates member-side prop's javadoc and doesn't seem to be correct.

Reply:
Why do you think that it is not correct? 
From the discovery spi perspective, we did not change the behavior. The `null` results with `false` there. But we are using this as a hint in a new implementation.

Reply:
> Why do you think that it is not correct?

I've double checked current usage and it seems to be correct. For some reasons I didn't see how it's used in `ClusterDiscoveryServiceBuilder`.

> But we are using this as a hint in a new implementation.

Does it make sense to describe this hint behavior? Otherwise, this change seems to be totally uncovered in built-in documentation of the client.